icu4j/main/tests/core/src/com/ibm/icu/dev/test/number/NumberFormatterApiTest.java

```diff
@@ -2991,6 +2991,17 @@ public void roundingFraction() {
                 ULocale.ENGLISH,
                 1,
                 "1");
+
+        assertFormatSingle(
+                "Hide If Whole with Rounding Mode (ICU-21881)",
+                ".00/w rounding-mode-floor",
+                ".00/w rounding-mode-floor",
+                NumberFormatter.with().precision(Precision.fixedFraction(2)
+                    .trailingZeroDisplay(TrailingZeroDisplay.HIDE_IF_WHOLE))
+                    .roundingMode(RoundingMode.FLOOR),
+                ULocale.ENGLISH,
+                3.001,
+                "3");
```

Comment:
Could you add a test w/ RoundingMode.HALF_UP sicne that is the case reported in the bug [ICU-21881](https://unicode-org.atlassian.net/browse/ICU-21881)